export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+    'ignored_experiments', [], 'List of experiment names to ignore. Experiment '
+    'must be in "slice.site" form not "site_slice".')
+flags.DEFINE_bool(
+    'pretty_json', None, 'Add extra indenting to json output (for debugging).')
+flags.DEFINE_string(
+    'output_dir', EXPORT_DIR, 'Root directory of json output files.')
+flags.DEFINE_string(
+    'output', None, 'Name of json output file. Set automatically if not given.')
+flags.DEFINE_string(
+    'export_metrics', METRIC_MAP_CONF, 'File name with metric map. The metric '
+    'map defines canonical metric names for raw, metric names taken from '
+    'collectd RRD files.')
+flags.DEFINE_bool(
+    'verbose', False, 'Increase verbosity level.')
+flags.DEFINE_bool(
+    'show_nagios', False, 'Shows collectd-nagios commands to monitor metrics.')
+flags.DEFINE_bool(
+    'show_rrdfile', False, 'Shows the RRD files opened during export.')
+flags.DEFINE_bool(
+    'show_metric', False, 'Shows the canonical metric names during export.')
+flags.DEFINE_bool(
+    'show_rawmetric', False, 'Shows the raw metric name before translating to '
+    'the canonical name. This can be helpful to know before adding new metrics '
+    'to the export metrics configuration file.')
+flags.DEFINE_bool(
+    'show_skipped', False, 'Shows the raw metric names that are not exported. '
+    'This option may be helpful when adding new metrics to the export_metrics.')
+flags.DEFINE_bool(
+    'update', True, 'Update timestamps on successful export. Update is always '
+    'disabled when any --show_* option is enabled.')
+flags.DEFINE_bool(
+    'compress', False,
+    'Compresses output and adds .gz extension to output filename.')
+
+
+class Error(Exception):
+  """Base error type for this file."""
+
+
+class TimeOptionError(Error):
+  """An error related to export times or ranges."""
+
+
+def init_global():
+  global HOSTNAME
+  # NOTE: This should be the hostname of root context, not slice context.
+  HOSTNAME = socket.gethostname()
+  logging.basicConfig(format='%(message)s', level=logging.INFO)
+
+
+class LastTime(object):
+  """Manages interactions with export timestamp."""
+
+  def __init__(self, filename):
+    self._filename = filename
+    self._lock_handle = None
+    self._first_creation = not os.path.exists(self._filename)
+
+  def __enter__(self):
+    self.open()
+    return self
+
+  def __exit__(self, *unused_args):
+    self.close()
+
+  def open(self):
+    """Opens the timestamp file, and creates the file if it does not exist."""
+    if self._first_creation:
+      # Since the file does not exist, use mode 'w' to create it.
+      self._lock_handle = open(self._filename, 'w')
+    else:
+      # File *does* exist, so use mode 'r' to preserve the mtime on the file.
+      self._lock_handle = open(self._filename, 'r')
+
+  def close(self):
+    """Closes the timestamp file."""
+    if self._lock_handle is not None:
+      self._lock_handle.close()
+
+  def get_mtime(self):
```

Comment:
First - I didn't know you could compare across arbitrary commits in Github. That's awesome and super useful!

I have some notes on the current implementation, but I think it will get buried if I do them here, so I'll comment on the latest changes.